src/filedata.cc

```diff
@@ -21,3561 +21,565 @@
 
 #include "filedata.h"
 
-#include <dirent.h>
-#include <sys/stat.h>
-#include <unistd.h>
-
-#include <cerrno>
-#include <cstdio>
-#include <cstdlib>
-#include <cstring>
-#include <ctime>
-#include <utility>
-
-#include <glib-object.h>
-#include <grp.h>
-#include <pwd.h>
-
-#include <config.h>
-
-#include "cache.h"
-#include "debug.h"
-#include "exif.h"
-#include "filefilter.h"
-#include "histogram.h"
-#include "intl.h"
-#include "main-defines.h"
-#include "main.h"
-#include "metadata.h"
-#include "misc.h"
-#include "options.h"
-#include "secure-save.h"
-#include "thumb-standard.h"
-#include "trash.h"
-#include "ui-fileops.h"
-
-#ifdef DEBUG_FILEDATA
-gint global_file_data_count = 0;
-#endif
-
-static GHashTable *file_data_pool = nullptr;
-static GHashTable *file_data_planned_change_hash = nullptr;
-
-static gint sidecar_file_priority(const gchar *extension);
-static void file_data_check_sidecars(const GList *basename_list);
-static void file_data_disconnect_sidecar_file(FileData *target, FileData *sfd);
-
-
-static SortType filelist_sort_method = SORT_NONE;
-static gboolean filelist_sort_ascend = TRUE;
-static gboolean filelist_sort_case = TRUE;
-
-/*
- *-----------------------------------------------------------------------------
- * text conversion utils
- *-----------------------------------------------------------------------------
- */
-
 gchar *text_from_size(gint64 size)
 {
-	gchar *a;
-	gchar *b;
-	gchar *s;
-	gchar *d;
-	gint l;
-	gint n;
-	gint i;
-
-	/* what I would like to use is printf("%'d", size)
-	 * BUT: not supported on every libc :(
-	 */
-	if (size > G_MAXINT)
-		{
-		/* the %lld conversion is not valid in all libcs, so use a simple work-around */
-		a = g_strdup_printf("%d%09d", static_cast<guint>(size / 1000000000), static_cast<guint>(size % 1000000000));
-		}
-	else
-		{
-		a = g_strdup_printf("%d", static_cast<guint>(size));
-		}
-	l = strlen(a);
-	n = (l - 1)/ 3;
-	if (n < 1) return a;
-
-	b = g_new(gchar, l + n + 1);
-
-	s = a;
-	d = b;
-	i = l - n * 3;
-	while (*s != '\0')
-		{
-		if (i < 1)
-			{
-			i = 3;
-			*d = ',';
-			d++;
-			}
-
-		*d = *s;
-		s++;
-		d++;
-		i--;
-		}
-	*d = '\0';
-
-	g_free(a);
-	return b;
+	return FileData::text_from_size(size);
```

Comment:
Not sure `text_from_*` belongs to `FileData` but it's better to leave it there for now.

Reply:
Agreed.  I think there are a lot of util functions that have ended up scattered around the codebase, and refactors like this one will make it easier to spot them and to move them to more appropriate homes.  I think it'll be best to evaluate that once filedata.cc is split into separate files, though.